conftest.py

```diff
@@ -54,82 +75,73 @@ def pytest_addoption(parser: Any) -> None:
-    global miniircd_procs
-    for m in miniircd_procs:
-        m.kill()
-    #shut down bitcoin and remove the regtest dir
-    local_command([os.path.join(bitcoin_path, "bitcoin-cli"), "-regtest",
-                    "-rpcuser=" + bitcoin_rpcusername,
-                   "-rpcpassword=" + bitcoin_rpcpassword, "stop"])
-    #note, it is better to clean out ~/.bitcoin/regtest but too
-    #dangerous to automate it here perhaps
+                     type="int",
+                     action="store",
+                     default=1,
+                     help="the number of local miniircd instances")
 
 
 @pytest.fixture(scope="session", autouse=True)
-def setup(request) -> None:
-    request.addfinalizer(teardown)
-
-    global bitcoin_conf, bitcoin_path, bitcoin_rpcpassword, bitcoin_rpcusername
-    bitcoin_path = request.config.getoption("--btcroot")
-    bitcoin_conf = request.config.getoption("--btcconf")
-    print("Here is the bitcoin_conf path:")
-    print(bitcoin_conf)
-    bitcoin_rpcpassword = request.config.getoption("--btcpwd")
-    bitcoin_rpcusername = request.config.getoption("--btcuser")
-
-    #start up miniircd
-    #minor bug in miniircd (seems); need *full* unqualified path for motd file
+def miniircd(pytestconfig):
+    """
+    Setup miniircd and handle its clean up.
+    """
+    # minor bug in miniircd (seems); need *full* unqualified path for motd file
+    miniircd_procs = []
     cwd = os.getcwd()
-    n_irc = request.config.getoption("--nirc")
-    global miniircd_procs
+    n_irc = pytestconfig.getoption("--nirc")
+    miniircd_path = os.path.join(root_path(), 'miniircd', 'miniircd')
+    motd_path = os.path.join(cwd, 'miniircd', 'testmotd')
     for i in range(n_irc):
-        miniircd_proc = local_command(
-            ["./miniircd/miniircd", "--ports=" + str(16667+i),
-             "--motd=" + cwd + "/miniircd/testmotd"],
-            bg=True)
+        command = f"{miniircd_path} --ports={str(16667 + i)} --motd={motd_path}"
+        miniircd_proc = local_command(command, bg=True)
         miniircd_procs.append(miniircd_proc)
+    yield None
+    # didn't find a stop command in miniircd, so just kill
+    for m in miniircd_procs:
+        m.kill()
 
-    # determine bitcoind version
-    bitcoind_version_string = subprocess.check_output([
-        os.path.join(bitcoin_path, "bitcoind"), "-version"]).split(b'\n')[0]
-    bitcoind_version = get_bitcoind_version(bitcoind_version_string)
-
-    #start up regtest blockchain
-    bitcoin_args = ["-regtest", "-daemon", "-conf=" + bitcoin_conf]
 
-    btc_proc = subprocess.call([os.path.join(bitcoin_path, "bitcoind")] +
-        bitcoin_args)
-    root_cmd = [os.path.join(bitcoin_path, "bitcoin-cli"), "-regtest",
-                       "-rpcuser=" + bitcoin_rpcusername,
-                       "-rpcpassword=" + bitcoin_rpcpassword]
+@pytest.fixture(scope="session", autouse=True)
+def regtest_bitcoind(pytestconfig):
```

Comment:
I think this would be better function name.
```suggestion
def setup_regtest_bitcoind(pytestconfig):
```